experimental/table/server/master/src/main/java/alluxio/master/catalog/AlluxioCatalog.java

```diff
@@ -159,6 +163,39 @@ private Database getDatabaseByName(String dbName) throws NotFoundException {
         .filter(info -> colNames.contains(info.getColName())).collect(Collectors.toList());
   }
 
+  /**
+   * Returns the statistics for the specified table.
+   *
+   * @param dbName the database name
+   * @param tableName the table name
+   * @param partNames partition names
+   * @param colNames column names
+   * @return the statistics for the partitions for a specific table
+   */
+  public Map<String, ColumnStatisticsList> getPartitionColumnStatistics(String dbName,
+      String tableName, List<String> partNames, List<String> colNames) throws IOException {
+    Table table = getTable(dbName, tableName);
+    List<Partition> partitions = table.getPartitions();
+    return partitions.stream().filter(p -> {
+      try {
+        return partNames.contains(ProtoUtils.extractHiveLayout(p.toProto()).getPartitionName());
+      } catch (InvalidProtocolBufferException e) {
+        return false;
+      }
+    }).map(p -> {
+      try {
+        PartitionInfo info = ProtoUtils.extractHiveLayout(p.toProto());
+        return new Pair<>(info.getPartitionName(), info.getColStats()
+            .getStatisticsList().stream().filter(e -> colNames.contains(e.getColName()))
+            .collect(Collectors.toList()));
+      } catch (InvalidProtocolBufferException e) {
+        return new Pair<>("", new ArrayList<ColumnStatisticsInfo>());
```

Comment:
does this denote empty/no statistics? Should it even be used by the caller?

Could we do a final pass on the result of this map, and then filter out entries with `("", ...)`?

Reply:
updated code no longer returns the empty entries.